marlowe-integration-tests/test/Language/Marlowe/Runtime/Integration/MarloweQuery.hs

```diff
@@ -0,0 +1,458 @@
+
+getTransactionsSpec :: SpecWith MarloweQueryTestData
+getTransactionsSpec = describe "getTransactions" do
+  for_ (Unknown : (Known <$> allContractNos)) \contractNo -> do
+    it (show contractNo) $ runMarloweQueryIntegrationTest \testData -> do
+      let
+        expected = SomeTransactions MarloweV1 <$> case contractNo of
+          Unknown -> Nothing
+          Known contractNo' -> Just $ contractNoToTransactions testData contractNo'
+      actual <- getTransactions $ contractNoToContractId testData contractNo
+      liftIO $ actual `shouldBe` expected
+
+getTransactionSpec :: SpecWith MarloweQueryTestData
+getTransactionSpec = describe "getTransaction" do
+  for_ (Unknown : ((Known . Left <$> allContractNos) <> (Known . Right <$> allTxNos))) \txNo -> do
+    it (show txNo) $ runMarloweQueryIntegrationTest \testData -> do
+      let
+        expected = case txNo of
+          Unknown -> Nothing
+          Known (Left _) -> Nothing
+          Known (Right txNo') -> Just $ txNoToSomeTransaction testData txNo'
+      actual <- getTransaction $ contractOrTxNoToTxId testData txNo
+      liftIO $ actual `shouldBe` expected
+
+getWithdrawalsSpec :: SpecWith MarloweQueryTestData
+getWithdrawalsSpec = describe "getWithdrawals" do
+  for_ (Unknown : (Known <$> allContractNos)) \contractNo -> do
+    it (show contractNo) $ runMarloweQueryIntegrationTest \testData -> do
+      let
+        expected = case contractNo of
+          Unknown -> Nothing
+          Known contractNo' -> Just $ contractNoToWithdrawals testData contractNo'
+      actual <- getWithdrawals $ contractNoToContractId testData contractNo
+      liftIO $ actual `shouldBe` expected
+
+getWithdrawalSpec :: SpecWith MarloweQueryTestData
+getWithdrawalSpec = describe "getWithdrawal" do
+  it "Returns Nothing for a fake txId" $ runMarloweQueryIntegrationTest \_ -> do
+    actual <- getWithdrawal "0000000000000000000000000000000000000000000000000000000000000000"
+    liftIO $ actual `shouldBe` Nothing
+  it "Returns Nothing for a create txId" $ runMarloweQueryIntegrationTest \testData -> do
+    actual <- getWithdrawal $ contractOrTxNoToTxId testData $ Known $ Left Contract1
+    liftIO $ actual `shouldBe` Nothing
+  it "Returns Nothing for an applyInputs txId" $ runMarloweQueryIntegrationTest \testData -> do
+    actual <- getWithdrawal $ contractOrTxNoToTxId testData $ Known $ Right Contract1Step4
+    liftIO $ actual `shouldBe` Nothing
+  it "Returns Just for a withdrawal txId" $ runMarloweQueryIntegrationTest \testData@MarloweQueryTestData{..} -> do
+    actual <- getWithdrawal $ fromCardanoTxId $ getTxId $ fst contract1Step5
+    liftIO $ actual `shouldBe` Just (contract1Step5Withdrawal testData)
+
+setup :: ActionWith MarloweQueryTestData -> IO ()
+setup runSpec = withLocalMarloweRuntime $ runIntegrationTest do
+  runtime <- ask
+  wallet1 <- getGenesisWallet 0
+  wallet2 <- getGenesisWallet 1
+  contract1 <- createStandardContract wallet1 wallet2
```

Comment:
So far as I can tell, the integration tests never create contracts that have stake credentials. On `mainnet` it will be more common than not that the Marlowe contract address will have a stake credential. Can we review `createStandardContract` so there is some probability of the contract address having a stake credential? Do we need a separate Jira ticket for this, or can it be done in this PR? Thanks.

Reply:
`createStandardContract` can already be created with addresses using stake credentials - it depends on how the `wallet`s get created.